webhook/webhook.go

```diff
@@ -329,30 +328,43 @@ func (ac *AdmissionController) register(
 	logger := logging.FromContext(ctx)
 	failurePolicy := admissionregistrationv1beta1.Fail
 
-	resources := sort.StringSlice{}
-	for k := range ac.Handlers {
+	var rules []admissionregistrationv1beta1.RuleWithOperations
+	for gvk := range ac.Handlers {
 		// Lousy pluralizer
-		resources = append(resources, strings.ToLower(k)+"s")
+		plural := strings.ToLower(gvk.Kind) + "s"
+
+		rules = append(rules, admissionregistrationv1beta1.RuleWithOperations{
+			Operations: []admissionregistrationv1beta1.OperationType{
+				admissionregistrationv1beta1.Create,
+				admissionregistrationv1beta1.Update,
+			},
+			Rule: admissionregistrationv1beta1.Rule{
+				APIGroups:   []string{gvk.Group},
+				APIVersions: []string{gvk.Version},
+				Resources:   []string{plural},
+			},
+		})
 	}
-	resources.Sort()
+
+	// Sort the rules by Group, Version, Kind so that things are deterministically ordered.
+	sort.Slice(rules, func(i, j int) bool {
+		lhs, rhs := rules[i], rules[j]
+		if lhs.APIGroups[0] != rhs.APIGroups[0] {
```

Comment:
`...[0]` looks like a bug in the waiting?

Reply:
Possibly, but all of the singletons are built a handful of lines above.